File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.c
#include <stdio.h>
#include <string.h>

int cnt = 0;

int recursion(char* s, int l, int r) {
	cnt++;
	if (l >= r) return 1;
	else if (s[l] != s[r]) return 0;
	else return recursion(s, l + 1, r - 1);
}

int isPalindrome(char* s) {
	cnt = 0;
	return recursion(s, 0, strlen(s) - 1);
}


int main(void) {
	int n;
	scanf("%d", &n);

	char arr[1001];
	while (n--) {
		scanf("%s", arr);
		printf("%d ", isPalindrome(arr));
		printf("%d\n", cnt);
	}
}

File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.py
import sys

cnt = 0
def recursion(s, l, r):
    global cnt
    cnt += 1
    if(l >= r): return 1
    elif(s[l] != s[r]): return 0
    else: return recursion(s, l + 1, r - 1)

def isPalindrome(s):
    global cnt
    cnt = 0
    return recursion(s, 0, len(s) - 1)

n = int(sys.stdin.readline())
while(n > 0):
    s = sys.stdin.readline().rstrip()
    print(isPalindrome(s), cnt)
    n -= 1